src/layout/umap.c

```diff
@@ -94,31 +94,91 @@ static igraph_error_t igraph_i_umap_find_sigma(const igraph_vector_t *distances,
     return IGRAPH_SUCCESS;
 }
 
+/* Check the distances argument, which should be NULL or a vector of nonnegative numbers */
+static igraph_error_t igraph_i_umap_check_distances(const igraph_vector_t *distances, igraph_integer_t no_of_edges) {
+
+    if (distances == NULL) {
+        return IGRAPH_SUCCESS;
+    }
+
+    if (igraph_vector_size(distances) != no_of_edges) {
+        IGRAPH_ERROR("Distances must be the same number as the edges in the graph.", IGRAPH_EINVAL);
+    }
+
+    for (igraph_integer_t eid = 0; eid != no_of_edges; eid++) {
```

Comment:
We have `igraph_vector_all_ge()` so you can use that instead of rolling your own loop. (It is also NaN-aware).

Reply:
Thanks. Hmm, that would require allocating a vector of the same length full of zeros though, wouldn't that be slower than this one?

Reply:
@iosonofabio Use `igraph_vector_min()`, see how it's done in `igraph_closeness()` for reference.

Reply:
I don't want the min though, I want to quit as soon as a negative number is found, which is faster.

Reply:
Is it necessary to optimize error conditions?  Error handling is already slow. This is true not only for igraph, but for almost any system, including the exception handling of most languages.

The pattern I referenced is used everywhere else in the library.

Reply:
Well, I just don't see why this needs to change. What if there's a NaN? What will the min function do? And will that be clear to the next person reading the code or do we have to add a comment explaining what happens in that case?

Whereas like this it's crystal clear...

Reply:
> Well, I just don't see why this needs to change.

Leave it as it is then.



> What if there's a NaN?

Please check the function I referenced (`igraph_closeness()`), and it will be clear to you.



> And will that be clear to the next person reading the code

That is precisely the point. People read code in patterns. This check is done identically in all other igraph functions, _except_ this one. When I see your code, I have to take a moment to check if it's correct. If I see the same pattern that I've seen many times before, I know it's fine, and I don't have to stop to think about it.

Reply:
I now changed it to match other functions and moved it to the body of the caller, for simplicity. Thanks for the suggestion.